huaweicloud/services/dli/resource_huaweicloud_dli_flinkjar_job.go

```diff
@@ -0,0 +1,512 @@
+		d.Set("manager_cu_number", detail.JobConfig.ManagerCuNumber),
+		d.Set("tm_cus", detail.JobConfig.TmCus),
+		d.Set("tm_slot_num", detail.JobConfig.TmSlotNum),
+		d.Set("resume_checkpoint", detail.JobConfig.ResumeCheckpoint),
+		d.Set("resume_max_num", detail.JobConfig.ResumeMaxNum),
+		d.Set("checkpoint_path", detail.JobConfig.CheckpointPath),
+		d.Set("runtime_config", utils.TagsToMap(parseConfig(detail.JobConfig.RuntimeConfig))),
+		d.Set("status", detail.Status),
+	)
+	if setSdErr := mErr.ErrorOrNil(); setSdErr != nil {
+		return fmtp.DiagErrorf("Error setting vault fields: %s", setSdErr)
+	}
+
+	return nil
+}
+
+func resourceFlinkJarJobDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.DliV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating DLI v1 client, err=%s", err)
+	}
+
+	jobId, aErr := strconv.Atoi(d.Id())
+	if aErr != nil {
+		return fmtp.DiagErrorf("the DLI flink job_id must be number. actual id=%s", d.Id())
+	}
+
+	deleteRst, dErr := flinkjob.Delete(client, jobId)
+	if dErr != nil {
+		return fmtp.DiagErrorf("delete DLI flink jar job failed. %q:%s", jobId, dErr)
+	}
+	if deleteRst != nil && !deleteRst.IsSuccess {
+		return fmtp.DiagErrorf("delete DLI flink jar job failed. %q:%s", jobId, dErr)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func resourceFlinkJarJobUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.DliV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating DLI v1 client, err=%s", err)
+	}
+
+	jobId, iErr := strconv.Atoi(d.Id())
+	if iErr != nil {
+		return fmtp.DiagErrorf("the DLI flink job_id must be number. actual id=%s", d.Id())
+	}
+
+	aErr := authorizeObsBucket(client, d)
+	if aErr != nil {
+		return aErr
+	}
+
+	uDiagErr := updateFlinkJarJobInRunning(client, jobId, d)
+	if uDiagErr != nil {
+		return uDiagErr
+	}
+
+	uDiagErr = updateFlinkJarJobWithStop(ctx, client, jobId, d)
+	if uDiagErr != nil {
+		return uDiagErr
+	}
+
+	checkCreateErr := checkFlinkJobRunResult(ctx, client, jobId, d.Timeout(schema.TimeoutCreate))
+	if checkCreateErr != nil {
+		return diag.FromErr(checkCreateErr)
+	}
+	return resourceFlinkJarJobRead(ctx, d, meta)
+}
+
+//updated in "job_running": smn_topic,restart_when_exception,resume_checkpoint,resume_max_num,checkpoint_path,obs_bucket
+func updateFlinkJarJobInRunning(client *golangsdk.ServiceClient, jobId int, d *schema.ResourceData) diag.Diagnostics {
+	if d.HasChanges("smn_topic", "restart_when_exception", "resume_checkpoint", "resume_max_num", "obs_bucket",
+		"checkpoint_path") {
+
+		opts := flinkjob.UpdateJarJobOpts{
+			ObsBucket:            d.Get("obs_bucket").(string),
+			SmnTopic:             d.Get("smn_topic").(string),
+			RestartWhenException: utils.Bool(d.Get("restart_when_exception").(bool)),
```

Comment:
I think this function should be moved to the golangsdk where `MaybeString` and `IntToPointer` are located.

Reply:
keep it as it is.